etl_entities/hwm_store/hwm_store_detect.py

```diff
@@ -0,0 +1,170 @@
+#  Copyright 2023 MTS (Mobile Telesystems)
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from __future__ import annotations
+
+from functools import wraps
+from typing import Any, Callable, Mapping
+
+from etl_entities.hwm_store.hwm_store_class_registry import HWMStoreClassRegistry
+
+
+def parse_config(value: Any, key: str) -> tuple[str, list, Mapping]:
+    if not isinstance(value, (str, Mapping)):
+        raise ValueError(f"Wrong value {value!r} for {key!r} config item")
+
+    store_type = "unknown"
+    args: list[Any] = []
+    kwargs: Mapping[str, Any] = {}
+
+    if isinstance(value, str):
+        return value, args, kwargs
+
+    for item in HWMStoreClassRegistry.known_types():
+        if item not in value:
+            continue
+
+        store_type = item
+        child = value[item]
+
+        args, kwargs = parse_child_item(child)
+
+    return store_type, args, kwargs
+
+
+def parse_child_item(child: Any) -> tuple[list, Mapping]:
+    store_args: list[Any] = []
+    store_kwargs: Mapping[str, Any] = {}
+
+    if not child:
+        return store_args, store_kwargs
+
+    if isinstance(child, str):
+        store_args = [child]
+    elif isinstance(child, Mapping):
+        store_kwargs = child
+    else:
+        store_args = child
```

Comment:
```suggestion
    if isinstance(child, Mapping):
        store_kwargs = child
    elif isinstance(child, Collection) and not isinstance(child, str):
        store_args = child
    else:
        store_args = [child]
```
Давай попробуем так. А то сейчас, если конфиг выглядит так:
```yml
my_hwm_store: 123
```
Это будет воспринято как `MyHWMStore(*123)`, и все упадет.